code/obj/item/plants.dm

```diff
@@ -437,15 +437,31 @@
 	desc = "By any other name, would smell just as sweet. This one likes to be called "
 	icon_state = "rose"
 	var/thorned = 1
-	var/list/names = list("Emma", "Olivia", "Ava", "Isabella", "Sophia", "Charlotte", "Mia", "Amelia",
-	"Harper", "Evelyn", "Abigail", "Emily", "Elizabeth", "Mila", "Dakota", "Avery",
-	"Sofia", "Camila", "Aria", "Scarlett", "Liam", "Noah", "William", "James",
-	"Oliver", "Benjamin", "Elijah", "Lucas", "Mason", "Logan", "Alexander", "Ethan",
-	"Jacob", "Michael", "Daniel", "Henry", "Jackson", "Sebastian", "Aiden", "Matthew")
 
 	New()
 		..()
-		desc = desc + pick(names) + "."
+		var/list/possible_names = list()
+		for(var/mob/living/carbon/human/H in mobs)
+			if(iswizard(H))
+				continue
+			if(isnukeop(H))
+				continue
+			if(!H.mind)
+				continue
+			possible_names += H
+		for(var/mob/living/silicon/ai/aimob in mobs)
+			possible_names += aimob
+		for(var/mob/living/silicon/robot/R in mobs)
+			if(!R.mind)
+				continue
+			possible_names += R
```

Comment:
You're essentially looping through `mobs` three times, as under the hood it's just doing `for(var/mob/M in mobs)` and then checking the types of the `M`s and only giving you those of said type. You could do just one loop with

```suggestion
		for(var/mob/M in mobs)
		        if(!M.mind)
		                continue
			if(ishuman(M))
				if(iswizard(M))
					continue
				if(isnukeop(M))
					continue
			  	possible_names += M
			else if(isAI(M) || isrobot(M))
			        possible_names += M
```

Reply:
Thanks! I'm still new to this and learning as I go, so thanks for the advice!